pkg/db/db.go

```diff
@@ -55,6 +54,30 @@ func Init(db *Database) {
 // Client is an interface for exposing some operations of managing database
 // client.
 type Client interface {
+        CreateFileShare(ctx *c.Context, fshare *model.FileShareSpec) (*model.FileShareSpec, error)
+
+        ListFileShares(ctx *c.Context) ([]*model.FileShareSpec, error)
+
+        ListFileSharesWithFilter(ctx *c.Context, m map[string][]string) ([]*model.FileShareSpec, error)
+
+        GetFileShare(ctx *c.Context, fshareID string) (*model.FileShareSpec, error)
+
+        UpdateFileShare(ctx *c.Context, fshare *model.FileShareSpec) (*model.FileShareSpec, error)
```

Comment:
we should allow user to modify with fileshareid and are should be allowed to modify name and description nothing else should be allowed to change, so you have to add fileshare id parameter instead of FileShareSpec

Reply:
Agree